370-auth-ui/oidc.md

```diff
@@ -0,0 +1,455 @@
+    access_token_lifetime: 1800
+    session_idle_timeout_enabled: true
+    session_idle_timeout: 300
+    client_name: Mobile App
+    redirect_uris:
+    - "https://example.com"
+    logo_uri: "https://example.com/logo.png"
+    grant_types:
+    - "authorization_code"
+    - "refresh_token"
+    response_types:
+    - "code"
+```
+
+### Parameters
+
+#### OIDC client Metadata
+
+Some parameters are defined in OIDC, see [ClientMetadata](https://openid.net/specs/openid-connect-registration-1_0.html#ClientMetadata).
+
+They are
+
+- `client_name`
+- `redirect_uris`
+- `login_uri`
+- `grant_types`
+- `response_types`
+
+#### Skygear-specific
+
+- `client_id`: OIDC client id.
+- `client_secret`: OIDC client secret, used in RP integration.
+- `access_token_lifetime`: OIDC access token lifetime in seconds, default to 1800.
+- `refresh_token_lifetime`: OIDC refresh token lifetime in seconds, default to max(access_token_lifetime, 86400). Must greater than or equal to `access_token_lifetime`.
+- `session_idle_timeout_enabled`: Indicate whether OIDC session idle timeout is enabled, default to `false`.
+- `session_idle_timeout`: The OIDC session idle timeout in seconds, default to min(`access_token_lifetime`, 300). Must less than or equal to `access_token_lifetime`.
+
+### Configuration in different cases
+
+#### Generic OIDC RP
+
+```yaml
+app_config:
+  clients:
+  - redirect_uris:
+    - "https://app-backend-endpoint.com"
+    grant_types:
+    - "authorization_code"
+    - "refresh_token"
+    response_types:
+    - "code"
+```
+
+- Should include app backend endpoint in `redirect_uris` which handle code to access token exchange
+- Generic OIDC RP run authentication code flow, and token endpoint will return refresh token. So `grant_types` should be [`authorization_code`, `refresh_token`] and `response_types` should be [`code`].
+
+#### Skygear Microservice case 1: Native app
+
+```yaml
+app_config:
+  clients:
+  - redirect_uris:
+    - "https://client-app-endpoint.com"
+    grant_types:
+    - "authorization_code"
+    - "refresh_token"
+    response_types:
+    - "code"
+```
+
+- Should include client app endpoint in `redirect_uris` which handle code to access token exchange
+- Native app run authentication code flow, and token endpoint will return refresh token. So `grant_types` should be [`authorization_code`, `refresh_token`] and `response_types` should be [`code`].
+
+
+#### Skygear Microservice case 3: First party web app
+
+```yaml
+app_config:
+  clients:
+  - redirect_uris:
+    - "https://client-app-endpoint.com"
+    grant_types: []
+    response_types:
+    - "none"
+```
+
+- Should include client app endpoint in `redirect_uris`. When Auth Gear set the Idp Session, it will redirect back to client app with empty result. Only authorized uris can be redirected.
+- First party web app use Idp session for authentication, no OIDC session will be created. So `grant_types` should be [] and `response_types` should be ["none"].
```

Comment:
It should be 'no OIDC grants would be returned', since there is no concept of session in OIDC.